platforms/documentation/docs/src/docs/userguide/dep-man/basics/declaring_dependencies.adoc

```diff
@@ -0,0 +1,225 @@
+Note:
+
+1. Gradle automatically includes _transitive dependencies_, which are dependencies of your dependencies.
+2. Gradle offers several  <<dependency_configurations.adoc#dependency-configurations,_configuration_>> options for dependencies, which define the scope in which dependencies are used, such as compile-time, runtime, or test-specific scenarios.
+3. You can specify <<declaring_repositories_adv.adoc#declaring-repositories,repositories>> where Gradle should look for dependencies in your build file.
+
+[[sec:dependency-types]]
+== Understanding types of dependencies
+
+There are two kinds of dependencies, module dependencies and file dependencies.
+
+[[sub:module_dependencies]]
+=== 1. Module dependencies
+
+Module dependencies are the most common dependencies.
+They refer to a module in a repository:
+
+[.multi-language-sample]
+=====
+[source, kotlin]
+.build.gradle.kts
+----
+dependencies {
+    implementation("org.springframework:spring-aop:2.5")
+    implementation("org.hibernate:hibernate:3.0.5")
+}
+----
+=====
+[.multi-language-sample]
+=====
+[source,groovy]
+.build.gradle
+----
+dependencies {
+    implementation('org.springframework:spring-core:2.5')
+    implementation('org.hibernate:hibernate:3.0.5')
+}
+----
+=====
+
+[[sub:file_dependencies]]
+=== 2. File dependencies
+
+In some projects, you might not rely on binary repository products like link:https://jfrog.com/artifactory/[JFrog Artifactory] or link:https://www.sonatype.com/products/sonatype-nexus-repository[Sonatype Nexus] for hosting and resolving external dependencies.
+Instead, you might host these dependencies on a shared drive or to check them into version control alongside the project source code.
+
+These are known as file dependencies because they represent files without any <<glossary.adoc#sub:terminology_module_metadata,metadata>> (such as information about transitive dependencies, origin, or author) attached to them.
+
+image::dependency-management-file-dependencies.png[]
+
+To add files as dependencies for a configuration, you simply pass a <<working_with_files.adoc#sec:file_collections,file collection>> as a dependency:
+
+====
+include::sample[dir="snippets/artifacts/externalDependencies/kotlin",files="build.gradle.kts[tags=file-dependencies]"]
+include::sample[dir="snippets/artifacts/externalDependencies/groovy",files="build.gradle[tags=file-dependencies]"]
+====
+
+== Looking at an example
+
+Let's imagine an example for a Java application which uses link:https://github.com/google/guava[Guava], a set of core Java libraries from Google:
+
+image::declaring-dependencies-2.png[]
+
+The Java app contains the following Java class:
+
+[source, java]
+.InitializeCollection.java
+----
+package com.myapp;
+
+import com.google.common.collect.ImmutableMap;  // Comes from guava library
+import java.util.Map;
+
+public class InitializeCollection {
+    public static void main(String[] args) {
+        Map items = ImmutableMap.of("coin", 3, "glass", 4, "pencil", 1);
```

Comment:
```suggestion
        Map<String, Int> items = ImmutableMap.of("coin", 3, "glass", 4, "pencil", 1);
```

This really feels like something that should live in a snippet and get auto-tested